systems/plants/@RigidBodyManipulator/geometricJacobian.m

```diff
@@ -45,8 +45,14 @@
     transformedMotionSubspaces = cellfun(@transformMotionSubspace, ...
       kinsol.T(joint_path), motionSubspaces, num2cell(signs'), 'UniformOutput', ...
       false); % change frame from body to world
-    J = cell2mat(transformedMotionSubspaces);
-    J = transformTwists(inv(kinsol.T{expressed_in}), J); % change frame from world to expressedIn
+  
+
+    for i = 1:1:length(transformedMotionSubspaces)
+       transformedMotionSubspaces{i} = transformedMotionSubspaces{i};
+    end
+
+    J = cell2matmsspoly(transformedMotionSubspaces);
```

Comment:
Not in the tests I've made.  It is possible I've missed a corner case, but
I haven't seen it break yet for any double entires.

-Andy S.

On Thu, Jul 30, 2015 at 11:11 PM, Twan Koolen <notifications@github.com>
wrote:

> In systems/plants/@RigidBodyManipulator/geometricJacobian.m
> <https://github.com/RobotLocomotion/drake/pull/1212#discussion_r35943757>:
>
> > @@ -45,8 +45,14 @@
> >      transformedMotionSubspaces = cellfun(@transformMotionSubspace, ...
> >        kinsol.T(joint_path), motionSubspaces, num2cell(signs'), 'UniformOutput', ...
> >        false); % change frame from body to world
> > -    J = cell2mat(transformedMotionSubspaces);
> > -    J = transformTwists(inv(kinsol.T{expressed_in}), J); % change frame from world to expressedIn
> > +
> > +
> > +    for i = 1:1:length(transformedMotionSubspaces)
> > +       transformedMotionSubspaces{i} = transformedMotionSubspaces{i};
> > +    end
> > +
> > +    J = cell2matmsspoly(transformedMotionSubspaces);
>
> does this break the method for non-msspoly inputs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/1212/files#r35943757>.
>